crates/host/src/wasmbus/mod.rs

```diff
@@ -938,121 +895,50 @@ impl Host {
-                } else {
-                    error!("data watch task unexpectedly stopped");
-                }
-                Ok(())
-            }
-        });
-
-        let heartbeat = spawn({
-            let host = Arc::clone(&host);
-            async move {
-                let mut heartbeat = Abortable::new(heartbeat, heartbeat_abort_reg);
-                heartbeat
-                    .by_ref()
-                    .for_each({
-                        let host = Arc::clone(&host);
-                        move |_| {
-                            let host = Arc::clone(&host);
-                            async move {
-                                let heartbeat = match host.heartbeat().await {
-                                    Ok(heartbeat) => heartbeat,
-                                    Err(e) => {
-                                        error!("failed to generate heartbeat: {e}");
-                                        return;
-                                    }
-                                };
-
-                                if let Err(e) =
-                                    host.publish_event("host_heartbeat", heartbeat).await
-                                {
-                                    error!("failed to publish heartbeat: {e}");
-                                }
-                            }
+                            // TODO graceful shutdown?
+                        _ = host.stopped() => {
+                            break;
                         }
-                    })
-                    .await;
-                let deadline = { *host.stop_rx.borrow() };
-                host.stop_tx.send_replace(deadline);
-                if heartbeat.is_aborted() {
-                    info!("heartbeat task gracefully stopped");
-                } else {
-                    error!("heartbeat task unexpectedly stopped");
+                    }
                 }
             }
         });
 
-        // Process existing data without emitting events
-        data.keys()
-            .await
-            .context("failed to read keys of lattice data bucket")?
-            .map_err(|e| anyhow!(e).context("failed to read lattice data stream"))
-            .try_filter_map(|key| async {
-                data.entry(key)
-                    .await
-                    .context("failed to get entry in lattice data bucket")
-            })
-            .for_each(|entry| async {
-                match entry {
-                    Ok(entry) => host.process_entry(entry, false).await,
-                    Err(err) => error!(%err, "failed to read entry from lattice data bucket"),
-                }
-            })
-            .await;
-
-        host.publish_event("host_started", start_evt)
-            .await
-            .context("failed to publish start event")?;
+        let lattice_names = host.lattices.keys().cloned().collect::<Vec<_>>();
+        for lattice in lattice_names.clone() {
```

Comment:
```suggestion
        for lattice in &lattice_names {
```

Reply:
This is cloned because it's used further down in the `async move` block.